src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -240,18 +240,21 @@ public bool IsDefinition
 
         /// <summary>
         /// <para>
-        /// Get a source location key for sorting. For performance, it's important that this be 
+        /// Get a source location key for sorting. For performance, it's important that this
         /// be able to be returned from a symbol without doing any additional allocations (even
         /// if nothing is cached yet.)
         /// </para>
         /// <para>
-        /// Only members of source namespaces, original source types, and namespaces that can be merged
-        /// need implement this function.
+        /// Only (original) source symbols and namespaces that can be merged
+        /// need implement this function if they want to do so for efficiency.
         /// </para>
         /// </summary>
         internal virtual LexicalSortKey GetLexicalSortKey()
```

Comment:
This used to be used for type symbols and namespaces only. Now it is also used for locals, range variables, and parameters. This implementation is good for most symbol kinds.